### ISSUE TITLE
[Bug]: Unable to attach server to two networks at first run

### ISSUE THREAD

**Issue by cebe** [hetznercloud/terraform-provider-hcloud#899]
### What happened?

I am trying to create 3 Servers and attach them to 2 separate networks.
This works when I create 3 servers attached to one network, then add the second network to the terraform definition and run terraform apply. It does not work on an empty project. Terraform apply shows the following error:

```
╷
│ Error: server is only allowed to be attached to each network once: 0
│ 
│   with hcloud_server.storage_mgmt[2],
│   on storage_servers.tf line 19, in resource "hcloud_server" "storage_mgmt":
│   19: resource "hcloud_server" "storage_mgmt" {
│ 
╵
```

### What did you expect to happen?

Behavior should be the same on an empty project and a partially deployed one.

### Please provide a minimal working example

```hcl
provider "hcloud" {
     token = ...
}

# use SSH key from home dir
resource "hcloud_ssh_key" "instance-ssh-key" {
  name       = "instance-ssh-key"
  public_key = file("~/.ssh/id_rsa.pub")
}

# network 1
resource "hcloud_network" "control-private" {
  name     = "control-1"
  ip_range = "10.200.0.0/16"
}
resource "hcloud_network_subnet" "control-storage-mgmt-subnet" {
  type         = "cloud"
  network_id   = hcloud_network.control-private.id
  network_zone = "eu-central"
  ip_range     = "10.200.222.0/24"
}

# network 2
resource "hcloud_network" "storage-private" {
  name     = "storage-1"
  ip_range = "10.222.0.0/16"
}
resource "hcloud_network_subnet" "storage-mgmt-subnet" {
  type         = "cloud"
  network_id   = hcloud_network.storage-private.id
  network_zone = "eu-central"
  ip_range     = "10.222.1.0/24"
}

resource "hcloud_server" "storage_mgmt" {
  count = 3
  name        = "mgmt${count.index + 1}.storage.test"
  image       = "debian-12"
  server_type = "cax11"
  location    = "fsn1"
  ssh_keys    = [hcloud_ssh_key.instance-ssh-key.id]
  # in case of upgrade, keep disk size to allow downgrade later
  keep_disk   = true

  public_net {
    ipv4_enabled = false
    ipv6_enabled = true
  }

  network {
    network_id = hcloud_network.storage-private.id
    ip = cidrhost(hcloud_network_subnet.storage-mgmt-subnet.ip_range, 10 + count.index)
  }

# network 2 (comment this to make the first run work, uncomment to get to the final state)
  network {
    network_id = hcloud_network.control-private.id
    ip = cidrhost(hcloud_network_subnet.control-storage-mgmt-subnet.ip_range, 10 + count.index)
  }

  connection {
    type        = "ssh"
    host        = self.ipv6_address
    user        = "root"
    private_key = file("~/.ssh/id_rsa")
  }

  depends_on = [
    hcloud_network_subnet.control-storage-mgmt-subnet,
    hcloud_network_subnet.storage-mgmt-subnet,
    hcloud_ssh_key.instance-ssh-key
  ]
}
```

**Comment by apricote** [comment]
Thanks for the report! 

It looks like this bug was added in https://github.com/hetznercloud/terraform-provider-hcloud/pull/594. The validation logic does not handle the case where the referenced networks will be created in the future and internally have an ID of `0`.

**Comment by apricote** [comment]
Will be fixed by #901. I plan on tagging a new release once the PR is merged.

**Comment by cebe** [comment]
Thanks for the quick response and fix!
